src/components/EditImageModal/index.jsx

```diff
@@ -539,6 +560,47 @@ export default class EditImageModal extends React.Component {
     return body;
   }
 
+  getImageSelectionModalBodyAssetsList = (type) => {
+    switch (type) {
+      case pageTypes.NORMAL:
+        return (<WrappedAssetsList />);
+      case pageTypes.NO_ASSETS:
+        return (
+          <div className="mt-3">
+            <WrappedMessage message={messages.editImageModalAssetsListNoAssetsMessage} tagName="h3" />
+          </div>
+        );
+      case pageTypes.NO_RESULTS:
+        return (
+          <React.Fragment>
+            <WrappedMessage message={messages.editImageModalAssetsListNoResultsMessage} tagName="h3" />
+            <WrappedAssetsClearSearchButton />
+          </React.Fragment>
+        );
+      case pageTypes.SKELETON:
```

Comment:
~~This is more of a question than a suggestion, but would it make sense to show the search bar while the Loading screen is shown? The reason I ask is because it's the AssetsList that's loading, not the actual search bar, right?~~

I realized this is just the initial modal loading state, right? It doesn't handle the loading state when a user has searched for something.

Reply:
Right, this is only shown when the modal is first opened and the request for assets is pending. I don't think the search, pagination, or result count components should show up in the skeleton state.